src/libraries/System.Net.Security/src/System/Net/Security/Pal.OSX/SafeDeleteSslContext.cs

```diff
@@ -155,48 +160,59 @@ protected override void Dispose(bool disposing)
 
         private unsafe int WriteToConnection(void* connection, byte* data, void** dataLength)
         {
-            ulong length = (ulong)*dataLength;
-            Debug.Assert(length <= int.MaxValue);
+            try
+            {
+                ulong length = (ulong)*dataLength;
+                Debug.Assert(length <= int.MaxValue);
 
-            int toWrite = (int)length;
-            var inputBuffer = new ReadOnlySpan<byte>(data, toWrite);
+                int toWrite = (int)length;
+                var inputBuffer = new ReadOnlySpan<byte>(data, toWrite);
 
-            _outputBuffer.EnsureAvailableSpace(toWrite);
-            inputBuffer.CopyTo(_outputBuffer.AvailableSpan);
-            _outputBuffer.Commit(toWrite);
+                _outputBuffer.EnsureAvailableSpace(toWrite);
+                inputBuffer.CopyTo(_outputBuffer.AvailableSpan);
+                _outputBuffer.Commit(toWrite);
+                // Since we can enqueue everything, no need to re-assign *dataLength.
 
-            // Since we can enqueue everything, no need to re-assign *dataLength.
-            const int noErr = 0;
-            return noErr;
+                return OSStatus_noErr;
+            }
+            catch
+            {
```

Comment:
Maybe add some trace logic here? I don't know if we care about that at this level.

Reply:
(similarly in the other function)

Reply:
I like that. I was originally thinking about saving exception and propagating it up but I think tracing is simple and should be sufficient for diagnostic.

Reply:
tracing added.